multiplayer/src/App.tsx

```diff
@@ -10,11 +13,29 @@ import "./App.css";
 function App() {
     const { state } = useContext(AppStateContext);
     const { signedIn } = state;
+    const [showSignInModal, setShowSignInModal] = useState(false);
```

Comment:
I can see the argument for moving the sign in/out apparatus here, but it means passing props to every component where you want to surface sign in functionality. Skillmap handles this by setting a state variable indicating which modal to show, and an app-level modal component to render whichever modal should be visible. That's not a bad system.

Let's try to render header bar as `<HeaderBar />`.